monitor/dbt_runner.py

```diff
@@ -56,13 +57,18 @@ def run_operation(self, macro_name: str, json_logs: bool = True, macro_args: dic
             json_args = json.dumps(macro_args)
             command_args.extend(['--args', json_args])
         success, command_output = self._run_command(command_args, json_logs)
+        if not success:
+            logger.error(f'Failed to run macro: "{macro_name}"')
         run_operation_results = []
         if json_logs:
             json_messages = command_output.splitlines()
             for json_message in json_messages:
                 log_message_dict = json.loads(json_message)
                 log_message_data_dict = log_message_dict.get('data')
                 if log_message_data_dict is not None:
+                    if log_message_dict['level'] == 'error':
```

Comment:
Are you sure 'level' always exists in log_message_dict? if you are not sure use 'get' 
Also, let's introduce this change more gradually so please add a param to the func called 'log_errors' that is disabled by default. Then in the unit-tests we will pass True there but in our users' deployments it won't be enabled for now by default.

Reply:
Yes, I'm confident that the `level` property is always present.  
The [`LogMessage`](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/logger.py#L50) class has a mandatory `level` field. I'll add the `logs_errors`.

Reply:
I'd still use get as dbt might change that in the future but I'll merge the PR and edit it before releasing the next version, thanks!

Reply:
There it is - [PR](https://github.com/elementary-data/dbt-data-reliability/pull/29).